tests/scattering1d/test_timefrequency_scattering.py

```diff
@@ -151,9 +151,111 @@ def test_frequency_scattering():
 
     # spinned=True
     X['coef'] = X['coef'].astype('complex64')
-    X['n'] = (-1, 4) # a mockup (n1, n2) pair from scattering1d_widthfirst
+    X['n'] = (-1, 4) # a mockup (n1, n2) pair from time_scattering_widthfirst
     freq_gen = frequency_scattering(X, S.backend, jtfs.filters_fr,
         jtfs.oversampling_fr, jtfs.average_fr=='local', spinned=True)
     for Y_fr in freq_gen:
         assert Y_fr['coef'].shape[-1] == X['coef'].shape[-1]
         assert Y_fr['n'] == (4, Y_fr['n_fr'])
+
+
+def test_joint_timefrequency_scattering():
+    # Define scattering object
+    J = 8
+    J_fr = 3
+    shape = (4096,)
+    Q = 8
+    S = TimeFrequencyScatteringBase(
+        J=J, J_fr=J_fr, shape=shape, Q=Q, T=0, F=0, backend='numpy')
+    S.build()
+    S.create_filters()
+    assert not S.average
+    assert not S.average_fr
```

Comment:
Not `S.average_local` and `S.average_local_fr`? These are the keywords used when creating the scattering object.

Reply:
`average_local` is not an attribute of `S` but simply the boolean `S.average == 'local'`. Same for `average_local_fr`

`not average_local` is equivalent to `self.average in [False, 'global']`

Reply:
Clear!